deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sDeployerUtil.java

```diff
@@ -104,4 +110,82 @@
         return K8sPodUtil.filterPods(podList, PodPredicates.podContainsLabel(), searchLabel);
 
     }
+    
+    /**
+     * Fetch service status from pod parent, called in case pods are not present
+     * @param context
+     * @return Not running status if pod owner found else not deployed(when service name provided)
+     */
+    public static List<DeploymentStatus> getOwnerDeploymentStatus(ApiClient apiClient, DeploymentContext context) {
+        if (context == null) {
+            return null;
+        }
+        logger.debug("No pods found, getting status from pods owner");
+        String serviceName = context.getServiceName();
+        String namespace = context.getNamespace();
```

Comment:
parent = getPodParent()
parent.getStatus() ?

Does this not solve the purpose ?